Src/DependencyCollector/Shared/Implementation/ClientServerDependencyTracker.cs

```diff
@@ -28,40 +28,51 @@ internal static DependencyTelemetry BeginTracking(TelemetryClient telemetryClien
             telemetry.Start();
             telemetryClient.Initialize(telemetry);
 #if NET45
-            // telemetry is initialized from current Activity (but not the Id)
-            // but every operation must have it's own Activity and Id must be set accordingly
-            // basically we repeat TelemetryClientExtensions.StartOperation here
-            var activity = new Activity(DependencyActivityName);
+            Activity activity;
+            Activity currentActivity = Activity.Current;
 
-            // This is workaround for the issue https://github.com/Microsoft/ApplicationInsights-dotnet/issues/538
-            // if there is no parent Activity, ID Activity generates is not random enough to work well with 
-            // ApplicationInsights sampling algorithm
-            // This code should go away when Activity is fixed: https://github.com/dotnet/corefx/issues/18418
-            if (Activity.Current == null)
+            // On .NET46 without profiler, outgoing requests are instrumented with reflection hook in DiagnosticSource 
+            //// see https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs
+            // it creates an Activity and injects standard 'Request-Id' and 'Correlation-Context' headers itself, so we should not start Activity in this case
+            if (currentActivity != null && currentActivity.OperationName == "System.Net.Http.Desktop.HttpRequestOut")
             {
-                activity.SetParentId(telemetry.Id);
+                activity = currentActivity;
             }
+            else
+            {
+                activity = new Activity(DependencyActivityName);
+
+                // This is workaround for the issue https://github.com/Microsoft/ApplicationInsights-dotnet/issues/538
+                // if there is no parent Activity, ID Activity generates is not random enough to work well with 
+                // ApplicationInsights sampling algorithm
+                // This code should go away when Activity is fixed: https://github.com/dotnet/corefx/issues/18418
+                if (Activity.Current == null)
+                {
+                    activity.SetParentId(telemetry.Id);
+                }
 
-            //// end of workaround
+                //// end of workaround
 
-            activity.Start();
-            
+                activity.Start();
```

Comment:
In case dependency was tracked with profiler, we need to generate new hierarchical Id for it. We use Activity to generate it.

We can also manually append AI-generated DependencyTelemetry.Id to the Activity.Current.Id (if any).
So the code would look a bit more simple.

``` 
            if (currentActivity != null && currentActivity.OperationName == "System.Net.Http.Desktop.HttpRequestOut")
            {
                telemetry.Id = currentActivity.Id;
            }            
            else 
            {
                var parentId = currentActivity?.Id ?? '|';
                telemetry.Id = parentId  + telemetry.Id + '.';
            }

            if (string.IsNullOrEmpty(telemetry.Context.Operation.Id))
            {
                telemetry.Context.Operation.Id = currentActivity?.RootId ??  telemetry.Id 
            }
```
 I'm fine with doing it this way